pages/en/lb3/Using-database-transactions.md

```diff
@@ -29,7 +29,80 @@ A LoopBack model can perform operations in a transaction when the model is atta
 
 ## Transaction APIs
 
-See the [API reference](http://apidocs.loopback.io/loopback-datasource-juggler/#transactionmixin) for full transaction API documentation.
+As of [v3.12.0](https://github.com/strongloop/loopback-datasource-juggler/tree/v3.12.0) of DataSource Juggler, there are two ways to handle transactions in LoopBack:
+
+* The [`TransactionMixin`](http://apidocs.loopback.io/loopback-datasource-juggler/#transactionmixin) gives access to the lower-level transaction API, leaving it up to the user to create and manage transaction objects, commit them on success or roll them back at the end of all intended operations.  
+  See [Lower-level Transaction API](#lower-level-transaction-api) below for more details.
+
+* To simplify and streamline the handling of transactions, [`dataSource.transaction()`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) offers a higher-level abstraction over the lower-level transaction API through a single method.  
+  See [Higher-level Transaction API](#higher-level-transaction-api) below for more details.
+
+## Higher-level Transaction API
+
+Inspired by how [Objection.js](http://vincit.github.io/objection.js/#binding-models-to-a-transaction) handles transactions, the method [`dataSource.transaction(execute, options, cb)`](https://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-transaction) creates transactions and binds them to models. Thus, you don't need to explicitly create a transaction object, passed it along to Model database methods, nor or do you need to commit it or roll it back at the end.
+
+Calling the `transaction(execute, options, cb)` method instead creates the transaction for you, then calls the `execute` function argument if it is provided and passes a `models` argument to it containing references to all models in this data-source. These model references are different from the ones in `app.models` in that they are automatically bound to the transaction that was just created. Smart caching is used to only created the bound models for the properties of the `models` argument that are actually accessed.
+
+At the end, `transaction.commit()` or `transaction.rollback()` is then automatically called, based on whether exceptions happen during execution or not. If no callback is provided to be called at the end of the execution, a Promise object is returned that is resolved or rejected as soon as the execution is completed, and the transaction is committed or rolled back.
+
+### Examples
+
+Here is how to create and commit a transaction:
+```js
+await app.dataSources.db.transaction(async models => {
+  const {MyModel} = models;
+  console.log(await MyModel.count()); // 0
+  await MyModel.create({foo: 'bar'});
+  console.log(await MyModel.count()); // 1
+})
```

Comment:
Nitpick: missing semicolon.

Reply:
Oops! I never write them, forgot that one. Will fix now. Shall I merge after?

Reply:
> Shall I merge after?

Yes please. Just squash the commits into a single one please, and wait for green CI results.